src/source-control/SourceControl.ts

```diff
@@ -39,6 +39,11 @@ export interface SourceControl {
    */
   getCommitByHash(hash: CommitHash): Promise<Commit | null>;
 
+  /**
+   * Get all the merge-base commit hashes on the long lived branches for a given commit 
+   */
+  getMergeCommitByCommitHash(commit: Commit) : Promise<CommitHash[]>;
```

Comment:
"Merge commit" typically means something else to most people I think. Maybe this will be more accurate?

```suggestion
  getMergeBasesWithAllLongLivedBranches(commit: Commit) : Promise<CommitHash[]>;
```